esphome/components/web_server/server_index.h

```diff
@@ -0,0 +1,572 @@
+#pragma once
+// Generated from https://github.com/esphome/esphome-webserver
```

Comment:
```suggestion
#include "esphome/core/hal.h"
// Generated from https://github.com/esphome/esphome-webserver
```

On platforms like esp-idf, `PROGMEM` is not globally defined

Reply:
Thanks for reviewing. Changes done